bolt-sidecar/src/api/commitments/firewall/processor.rs

```diff
@@ -0,0 +1,350 @@
+                    trace!(?rpc_url, ?commitment, "received commitment response");
+                    let message = match &commitment {
+                        SignedCommitment::Inclusion(ic) => {
+                            ic.into_identified(id);
+                            Message::text(
+                                serde_json::to_string(&ic)
+                                    .expect("to stringify inclusion commitment"),
+                            )
+                        }
+                    };
+
+                    // Add the message to the outgoing messages queue
+                    this.outgoing_messages.push_back(message);
+                }
+            }
+
+            // 2. Process outgoing messages
+            while let Some(message) = this.outgoing_messages.pop_front() {
+                // Check if the write sink is able to receive data.
+                match this.write_sink.poll_ready_unpin(cx) {
+                    Poll::Ready(Ok(())) => {
+                        progress = true;
+
+                        // Try to send the message to the sink, for later flushing.
+                        if let Err(e) = this.write_sink.start_send_unpin(message) {
+                            error!(?e, ?rpc_url, "failed to send message to websocket write sink");
+                            continue;
+                        }
+                    }
+                    Poll::Pending => {
+                        // Put the message back and try again later
+                        this.outgoing_messages.push_front(message);
+                        break;
+                    }
+                    Poll::Ready(Err(e)) => {
+                        error!(?e, "sink error while sending message to websocket");
+                        continue;
+                    }
+                }
+            }
+
+            // 3. Ensure the write sink is flushed so that message are sent to the caller server.
+            //
+            // NOTE: We're not considering "progress" flushing the sink, i.e. `Poll::Ready(())`.
+            // That is because flushing an empty sink would lead to run this loop indefinitely
+            if let Poll::Ready(Err(e)) = this.write_sink.poll_flush_unpin(cx) {
+                error!(?e, "failed to flush websocket write sink");
+                continue;
+            }
+
+            // 4. Handle shutdown signals before accepting any new work
+            match this.shutdown_rx.try_recv() {
+                Ok(_) => {
+                    info!("received shutdown signal. closing websocket connection...");
+                    return Poll::Ready(InterruptReason::Shutdown);
+                }
+                Err(TryRecvError::Closed) => {
+                    error!("shutdown channel closed");
+                }
+                _ => {}
+            }
+
+            // Incoming work tasks
+
+            // 5. Handle incoming websocket messages from the read stream.
```

Comment:
consider adding rate-limiting to protect against rogue firewalls

Reply:
The idea of firewall delegation is to not worry about rate-limiting and filtering, which is done by the trusted operator to which you're connected to. 
Can you provide an example about the second comment?

Reply:
I get the reasoning, let's think about it at a later stage! cc @mempirate

Reply:
Yeah I agree with Havard, we need to have a defense in depth mindset about these things. Trusting an external entity is fine until it's not, issues can arise not just from malicious intent but also hacks, bugs, and other stuff. Always remember that this is running on validator hardware.

We can create a follow-up issue though, doesn't need to be implemented right away.

Reply:
https://github.com/chainbound/bolt/issues/646